arc/interface.py

```diff
@@ -128,13 +131,29 @@ def fmt(self, colored=False, with_output=True) -> str:
 
     def as_np(self, with_solution=True):
         return (self.input.np, self.output.np if with_solution else None)
+    
+    def augment(self, augment_funcs: list[Callable]=None, with_solution: bool=True) -> 'BoardPair':
+        if not augment_funcs and not self.default_augment_funcs:
+            raise ValueError(f"No augmentation functions found. Pass augment_funcs to augment method or set default_augment_funcs")
+        augment_funcs = augment_funcs if augment_funcs else self.default_augment_funcs
+        augment_funcs = [func() for func in augment_funcs]
+        augmenter = partial(reduce, lambda acc,f: f(acc), augment_funcs)
+        return self._augment(augmenter,with_solution)
+    
+    def _augment(self, augmenter: Callable, with_solution: bool) -> 'BoardPair':
+        this = self if self.original is None else self.original
+        input = augmenter(this.input.data)
```

Comment:
it seems like this will change the type of `input` and `output` from `Board` to `list[list[int]]`. Could we either make this back into `Board` here, or have the augmentation functions output `Board`s?

Reply:
@jcole75 ping :)